src/engine/openspaceengine.cpp

```diff
@@ -497,22 +499,124 @@ void OpenSpaceEngine::initialize() {
     // Load a light and a monospaced font
     loadFonts();
 
-    // Initialize the Scene
-    // @CLEANUP:  This should become a unique_ptr that is either created inside the
-    // renderengine or moved into it ---abock
-    Scene* sceneGraph = new Scene;
-    sceneGraph->initialize();
-    
     std::string scenePath = "";
     configurationManager().getValue(ConfigurationManager::KeyConfigScene, scenePath);
-    sceneGraph->scheduleLoadSceneFile(scenePath);
 
-    // Initialize the RenderEngine
-    _renderEngine->setSceneGraph(sceneGraph);
     _renderEngine->initialize();
+
+    scheduleLoadScene(scenePath);
+
+    LINFO("Finished initializing");
+}
+
+void OpenSpaceEngine::scheduleLoadScene(const std::string& scenePath) {
```

Comment:
How will this potentially save a copy? Is this correct:
Const ref: No copy into the new stack frame. 1 copy from the reference to the member variable. Total 1 copy.
Pass by value and move semantics: One copy into the new stack frame. No copy into the member value (only swap pointers). Total 1 copy.

Semantically though, I guess one could argue that since the method is making a copy of the value anyway, it makes sense to show that explicitly in the signature, instead of hiding that in the implementation. However, you could also argue that the fact that the value is stored in the class is an implementation detail, and if we change the internal structure of the class, the API can be unchanged (while the copy is avoided) if we keep it as const std::string&.

:) I'll change it to std::string and std::move for now, but I think this is an interesting discussion